spring-rabbit-test/src/main/java/org/springframework/amqp/rabbit/test/context/SpringRabbitContextCustomizer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.test.context;
+
+import org.springframework.amqp.rabbit.annotation.RabbitBootstrapConfiguration;
+import org.springframework.amqp.rabbit.config.AbstractRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.config.DirectRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.config.SimpleRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.junit.BrokerRunningSupport;
+import org.springframework.amqp.rabbit.junit.RabbitAvailableCondition;
+import org.springframework.amqp.rabbit.test.context.SpringRabbitTest.ContainerType;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.test.context.ContextCustomizer;
+import org.springframework.test.context.MergedContextConfiguration;
+import org.springframework.util.Assert;
+
+/**
+ * Adds infrastructure beans to the Spring test context.
+ *
+ * @author Gary Russell
+ * @since 2.3
+ *
+ */
+class SpringRabbitContextCustomizer implements ContextCustomizer {
```

Comment:
Shouldn't it be SpringRabbit**Test**ContextCustomizer ?

Reply:
Why? It's already in a test jar and package.

I will change it if you insist.

Reply:
...and `ContextCustomizer` has no `Test`.

Reply:
Ah! OK. I can live with that. 😄